components/drop-zone/provider.js

```diff
@@ -17,6 +17,13 @@ class DropZoneProvider extends Component {
 		this.isWithinZoneBounds = this.isWithinZoneBounds.bind( this );
 		this.onDrop = this.onDrop.bind( this );
 
+		const throttledAction = throttle( this.toggleDraggingOverDocument, 500 );
+
+		this.dragOverListener = ( event ) => {
```

Comment:
Minor: I guess it's just a preference of mine but what do you think about something like this:

```js
this.toggleDraggingOverDocument = throttle( this.toggleDraggingOverDocument.bind( this ), 500 );
this.dragOverListener = this.dragOverListener.bind( this );
```

and move the `dragOverListener` declaration as a regular method (not defined in the constructor).

Reply:
Also wondering if `500` is not too high :) maybe 200?

Reply:
Indeed, `200` is a lot better.

> Minor: I guess it's just a preference of mine but what do you think about something like this:...

Definitely. That is a lot cleaner/better. Just updated now. :-) 

Also, you'll notice I make the call to the throttled method as `this.toggleDraggingOverDocument.call( this, event );` - it's not necessary - we can just `this.toggleDraggingOverDocument( event )` - lodash uses the last `this` value on each call. I don't know why but I feel safer to have it this way instead? Should I just revert to the normal invocation?

Reply:
Sorry this is going back/forth so many times... :-)

Reply:
No worries, thanks for handling all these details :)

I'd be in favor of a regular `this.toggleDraggingOverDocument( event )` 

I'll give it a last spin tomorrow morning and approve

Reply:
Awesome! Updated. Let me know if you spot anything else :-)